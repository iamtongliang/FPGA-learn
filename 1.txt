十进制 |原码 | 反码 |补码 
   7 | 0111 | 0111 | 0111  
   6 | 0110 | 0110 | 0110  
   5 | 0101 | 0101 | 0101  
   4 | 0100 | 0100 | 0100  
   3 | 0011 | 0011 | 0011  
   2 | 0010 | 0010 | 0010  
   1 | 0001 | 0001 | 0001  
   0 | 0000 | 0000 | 0000  
  -1 | 1001 | 1001 | 1111 
  -2 | 1010 | 1010 | 1110 
  -3 | 1011 | 1011 | 1101 
  -4 | 1100 | 1100 | 1100 
  -5 | 1101 | 1101 | 1011 
  -6 | 1110 | 1110 | 1010 
  -7 | 1111 | 1111 | 1001 
  -8 | 11000 | 11000 | 1000 
 