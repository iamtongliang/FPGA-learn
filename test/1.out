#! /usr/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 10;
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/system.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_sys.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/vhdl_textio.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/v2005_math.vpi";
:vpi_module "/usr/lib/x86_64-linux-gnu/ivl/va_math.vpi";
S_0x61457f4f2b20 .scope module, "tb_c2code" "tb_c2code" 2 7;
 .timescale -9 -10;
P_0x61457f4d6610 .param/l "CLK_PERIOD" 1 2 10, +C4<00000000000000000000000000001010>;
L_0x749c31f86060 .functor BUFT 1, C4<0>, C4<0>, C4<0>, C4<0>;
v0x61457f5221e0_0 .net *"_ivl_12", 0 0, L_0x749c31f86060;  1 drivers
L_0x749c31f86018 .functor BUFT 1, C4<0>, C4<0>, C4<0>, C4<0>;
v0x61457f5222e0_0 .net *"_ivl_7", 0 0, L_0x749c31f86018;  1 drivers
v0x61457f5223c0_0 .var "clk", 0 0;
v0x61457f522460_0 .var/i "i", 31 0;
v0x61457f522500_0 .var "ina", 7 0;
v0x61457f5225e0_0 .var "inb", 7 0;
v0x61457f5226c0_0 .net "outa", 7 0, L_0x61457f522c10;  1 drivers
v0x61457f5227a0_0 .net "outb", 7 0, L_0x61457f522d80;  1 drivers
L_0x61457f522a10 .part v0x61457f522500_0, 0, 7;
L_0x61457f522b10 .part v0x61457f5225e0_0, 0, 7;
L_0x61457f522c10 .concat [ 7 1 0 0], v0x61457f521a10_0, L_0x749c31f86018;
L_0x61457f522d80 .concat [ 7 1 0 0], v0x61457f521de0_0, L_0x749c31f86060;
S_0x61457f506830 .scope module, "inst1" "ComplimentCode" 2 30, 3 3 0, S_0x61457f4f2b20;
 .timescale -9 -10;
    .port_info 0 /INPUT 7 "A";
    .port_info 1 /INPUT 7 "B";
    .port_info 2 /OUTPUT 7 "Aout";
    .port_info 3 /OUTPUT 7 "Bout";
    .port_info 4 /INPUT 1 "clk1";
P_0x61457f506a10 .param/l "bitNumber" 0 3 4, +C4<00000000000000000000000000000111>;
v0x61457f506f90_0 .net "A", 6 0, L_0x61457f522a10;  1 drivers
v0x61457f5084b0_0 .var "Ainbuf", 6 0;
v0x61457f521950_0 .net "Aout", 6 0, v0x61457f521a10_0;  1 drivers
v0x61457f521a10_0 .var "Aoutbuf", 6 0;
v0x61457f521af0_0 .net "B", 6 0, L_0x61457f522b10;  1 drivers
v0x61457f521c20_0 .var "Binbuf", 6 0;
v0x61457f521d00_0 .net "Bout", 6 0, v0x61457f521de0_0;  1 drivers
v0x61457f521de0_0 .var "Boutbuf", 6 0;
v0x61457f521ec0_0 .var "Cinbuf", 1 0;
v0x61457f521fa0_0 .net "clk1", 0 0, v0x61457f5223c0_0;  1 drivers
v0x61457f522060_0 .var/i "i", 31 0;
E_0x61457f4ffd90/0 .event anyedge, v0x61457f506f90_0, v0x61457f521af0_0, v0x61457f5084b0_0, v0x61457f521c20_0;
E_0x61457f4ffd90/1 .event anyedge, v0x61457f521ec0_0, v0x61457f521a10_0, v0x61457f521de0_0;
E_0x61457f4ffd90 .event/or E_0x61457f4ffd90/0, E_0x61457f4ffd90/1;
    .scope S_0x61457f506830;
T_0 ;
    %wait E_0x61457f4ffd90;
    %load/vec4 v0x61457f506f90_0;
    %store/vec4 v0x61457f5084b0_0, 0, 7;
    %load/vec4 v0x61457f521af0_0;
    %store/vec4 v0x61457f521c20_0, 0, 7;
    %load/vec4 v0x61457f5084b0_0;
    %parti/s 1, 6, 4;
    %ix/load 4, 6, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521a10_0, 4, 1;
    %load/vec4 v0x61457f521c20_0;
    %parti/s 1, 6, 4;
    %ix/load 4, 6, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521de0_0, 4, 1;
    %load/vec4 v0x61457f5084b0_0;
    %parti/s 1, 6, 4;
    %pushi/vec4 0, 0, 1;
    %xor;
    %ix/load 4, 0, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521ec0_0, 4, 1;
    %load/vec4 v0x61457f521c20_0;
    %parti/s 1, 6, 4;
    %pushi/vec4 0, 0, 1;
    %xor;
    %ix/load 4, 1, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521ec0_0, 4, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x61457f522060_0, 0, 32;
T_0.0 ;
    %load/vec4 v0x61457f522060_0;
    %cmpi/s 6, 0, 32;
    %jmp/0xz T_0.1, 5;
    %load/vec4 v0x61457f5084b0_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521ec0_0;
    %parti/s 1, 0, 2;
    %xor;
    %ix/getv/s 4, v0x61457f522060_0;
    %store/vec4 v0x61457f521a10_0, 4, 1;
    %load/vec4 v0x61457f521c20_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521ec0_0;
    %parti/s 1, 1, 2;
    %xor;
    %ix/getv/s 4, v0x61457f522060_0;
    %store/vec4 v0x61457f521de0_0, 4, 1;
    %load/vec4 v0x61457f522060_0;
    %addi 1, 0, 32;
    %store/vec4 v0x61457f522060_0, 0, 32;
    %jmp T_0.0;
T_0.1 ;
    %pushi/vec4 0, 0, 1;
    %ix/load 4, 0, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521ec0_0, 4, 1;
    %pushi/vec4 0, 0, 1;
    %ix/load 4, 1, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521ec0_0, 4, 1;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x61457f522060_0, 0, 32;
T_0.2 ;
    %load/vec4 v0x61457f522060_0;
    %cmpi/s 6, 0, 32;
    %jmp/0xz T_0.3, 5;
    %load/vec4 v0x61457f521a10_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521a10_0;
    %parti/s 1, 6, 4;
    %xor;
    %load/vec4 v0x61457f521ec0_0;
    %parti/s 1, 0, 2;
    %xor;
    %ix/getv/s 4, v0x61457f522060_0;
    %store/vec4 v0x61457f521a10_0, 4, 1;
    %load/vec4 v0x61457f521de0_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521de0_0;
    %parti/s 1, 6, 4;
    %xor;
    %load/vec4 v0x61457f521ec0_0;
    %parti/s 1, 1, 2;
    %xor;
    %ix/getv/s 4, v0x61457f522060_0;
    %store/vec4 v0x61457f521de0_0, 4, 1;
    %load/vec4 v0x61457f521a10_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521a10_0;
    %parti/s 1, 6, 4;
    %and;
    %load/vec4 v0x61457f521ec0_0;
    %parti/s 1, 0, 2;
    %load/vec4 v0x61457f521a10_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521a10_0;
    %parti/s 1, 6, 4;
    %xor;
    %and;
    %or;
    %ix/load 4, 0, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521ec0_0, 4, 1;
    %load/vec4 v0x61457f521de0_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521de0_0;
    %parti/s 1, 6, 4;
    %and;
    %load/vec4 v0x61457f521ec0_0;
    %parti/s 1, 1, 2;
    %load/vec4 v0x61457f521de0_0;
    %load/vec4 v0x61457f522060_0;
    %part/s 1;
    %load/vec4 v0x61457f521de0_0;
    %parti/s 1, 6, 4;
    %xor;
    %and;
    %or;
    %ix/load 4, 1, 0;
    %flag_set/imm 4, 0;
    %store/vec4 v0x61457f521ec0_0, 4, 1;
    %load/vec4 v0x61457f522060_0;
    %addi 1, 0, 32;
    %store/vec4 v0x61457f522060_0, 0, 32;
    %jmp T_0.2;
T_0.3 ;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_0x61457f4f2b20;
T_1 ;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v0x61457f522500_0, 0, 8;
    %pushi/vec4 0, 0, 8;
    %store/vec4 v0x61457f5225e0_0, 0, 8;
    %end;
    .thread T_1;
    .scope S_0x61457f4f2b20;
T_2 ;
    %delay 50, 0;
    %load/vec4 v0x61457f5223c0_0;
    %inv;
    %store/vec4 v0x61457f5223c0_0, 0, 1;
    %jmp T_2;
    .thread T_2;
    .scope S_0x61457f4f2b20;
T_3 ;
    %vpi_call 2 14 "$dumpfile", "tb_c2code.vcd" {0 0 0};
    %vpi_call 2 15 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x61457f4f2b20 {0 0 0};
    %end;
    .thread T_3;
    .scope S_0x61457f4f2b20;
T_4 ;
    %pushi/vec4 0, 0, 32;
    %store/vec4 v0x61457f522460_0, 0, 32;
T_4.0 ;
    %load/vec4 v0x61457f522460_0;
    %cmpi/s 63, 0, 32;
    %flag_or 5, 4;
    %jmp/0xz T_4.1, 5;
    %delay 50, 0;
    %load/vec4 v0x61457f522460_0;
    %pad/s 8;
    %store/vec4 v0x61457f522500_0, 0, 8;
    %load/vec4 v0x61457f522460_0;
    %pushi/vec4 63, 0, 32;
    %and;
    %pushi/vec4 64, 0, 32;
    %or;
    %pad/u 8;
    %store/vec4 v0x61457f5225e0_0, 0, 8;
    %load/vec4 v0x61457f522460_0;
    %addi 1, 0, 32;
    %store/vec4 v0x61457f522460_0, 0, 32;
    %jmp T_4.0;
T_4.1 ;
    %delay 50, 0;
    %vpi_call 2 27 "$finish" {0 0 0};
    %end;
    .thread T_4;
# The file index is used to find the file name in the following table.
:file_names 4;
    "N/A";
    "<interactive>";
    "tb_c2code.v";
    "/home/liang-tong/learn/lang/verilog/test/complement2code.v";
